src/java/org/apache/cassandra/index/sai/disk/PostingList.java

```diff
@@ -60,4 +61,77 @@ default void close() throws IOException {}
      * @return first segment row ID which is >= the target row ID or {@link PostingList#END_OF_STREAM} if one does not exist
      */
     int advance(int targetRowID) throws IOException;
+
+    /**
+     * @return peekable wrapper of current posting list
+     */
+    default PeekablePostingList peekable()
+    {
+        return new PeekablePostingList(this);
+    }
+
+    public static class PeekablePostingList implements PostingList
```

Comment:
I see we also have a `MergePostingList#PeekingPostingList`. I wondering if we should reuse anything? Maybe not given their differences, just thinking out loud.

Reply:
Didn't notice `MergePostingList#PeekingPostingList`. I think it can be replaced by `PeekablePostingList`. The complexity was due to TPC async IO.

Reply:
I have simplified `MergePostingList` as we don't need to handle NotInCacheException